package/base-files/files/bin/config_generate

```diff
@@ -160,18 +160,18 @@ generate_network() {
 
 	case "$protocol" in
 		static)
-			local ipad
+			local ipad netm bitm
 			case "$1" in
 				lan) ipad=${ipaddr:-"192.168.1.1"} ;;
 				*) ipad=${ipaddr:-"192.168.$((addr_offset++)).1"} ;;
 			esac
 
 			netm=${netmask:-"255.255.255.0"}
+			bitm=$(bitcount $(ip2int ${netm}))
```

Comment:
I wouldn't call this `bitm` because that's not very clear.  `prefix` is a better name.

I'm not sure why it's not working for you, but I just ran a trivial program and it worked fine.

```
root@pigpen:~# cat /tmp/bar
#!/bin/sh

. /lib/functions/ipv4.sh

netmask="255.255.255.0"

prefix=$(bitcount $(ip2int $netmask))

echo "prefix=$prefix"
root@pigpen:~# sh /tmp/bar
prefix=24
root@pigpen:~# 
```

I'd try adding some debugging, like seeing what gets passed to `ip2int` and what the return value is.

Reply:
I changed the name as you suggested.
You misunderstood, this part works fine.

Reply:
See my question above: https://github.com/openwrt/openwrt/pull/13780#issuecomment-1793298602.

Why is it generating `dhcp-range=set:lan,0.0.0.1,0.0.0.254,0.0.0.0,12h`?  That 4th field is clearly wrong.  Quoting the `dnsmasq(8)` man page:

> This option may be repeated, with different addresses, to enable DHCP service to more than one network. For directly connected networks (ie, networks on which the machine running dnsmasq has an interface) the netmask is optional: dnsmasq will determine it from the interface configuration. For networks which receive DHCP service via a relay agent, dnsmasq cannot determine the netmask itself, so it should be specified, otherwise dnsmasq will have to guess, based on the class (A, B or C) of the network address. The broadcast address is always optional. It is always allowed to have more than one --dhcp-range in a single subnet.

Since we're talking about a directly connected server and *not* a relay, why are we even bothering with this field?  We should make it go away.

Reply:
Since this specific part works correctly, I mark it as resolved.
You can continue discussing dnsmasq related code here:
https://github.com/openwrt/openwrt/pull/13870